matchingengine/OrderBook.go

```diff
@@ -135,7 +135,20 @@ func (ob *orderBook) AddAskOrder(ask *models.Ask) {
 	// in case of stoploss order, add it to stoploss queue and return
 	if ask.OrderType == models.StopLoss {
 		l.Debugf("Adding stopLoss with ask_id %d to the queue", ask.Id)
-		ob.askStoploss.Push(ask)
+		stockCopy, _ := models.GetStockCopy(ask.StockId)
```

Comment:
Potential nil error. ask could have invalid `StockId`

Reply:
Ignore this if there's already a check before this method gets called (likely is, I might've simply got confused)

Reply:
StockId is not user input as far as I know. It is based on which stock the user wants to place an order for.

Reply:
By user input I meant anything that comes from the client request, not necessarily from the UI. 

But in this case, the stockID gets passed by the MatchingEngine layer, so it's fine. Sorry for the noise, just got a bit confused :P